ekan0ra/queue.py

```diff
@@ -27,13 +27,11 @@ def enqueue(self, nick):
 
     def dequeue(self, nick):
         """
-        Remove first item matching `nick` in queue.
+        Remove every occurrence of user from a queue.
         """
-        try:
-            self.remove(nick)
-            return True
-        except ValueError:
-            return False
+        result=self.count(nick)>0
+        self=filter(lambda x: x!=nick,self)
```

Comment:
Your implementation of this line is wrong. It won't work as expected.

**This should be:** `self[:] = filter(lambda x: x != nick, self)`